src/common/__tests__/__snapshots__/graphHelpers.test.js.snap

```diff
@@ -1,223 +1,261 @@
+Object {
+  "chartData": Array [
+    Object {
+      "tooltip": "5 t('curiosity-graph.tooltipLabel') Apr",
+      "x": 0,
+      "xAxisLabel": "Apr",
+      "y": 5,
+    },
+    Object {
+      "tooltip": "7 t('curiosity-graph.tooltipLabel') Aug
+ +2 t('curiosity-graph.tooltipPreviousLabelQuarterly')",
+      "x": 1,
+      "xAxisLabel": "Aug",
+      "y": 7,
+    },
+    Object {
+      "tooltip": "7 t('curiosity-graph.tooltipLabel') Dec",
+      "x": 2,
+      "xAxisLabel": "Dec",
+      "y": 7,
+    },
+    Object {
+      "tooltip": "7 t('curiosity-graph.tooltipLabel') Apr 2019",
+      "x": 3,
+      "xAxisLabel": "Apr
+2019",
+      "y": 7,
+    },
   ],
+  "chartXAxisLabelIncrement": 1,
 }
 `;
 
-exports[`GraphHelpers should get tick values array given chart data 1`] = `
+exports[`GraphHelpers should handle cross year labels: cross year 1`] = `
 Object {
-  "empty array": Array [],
-  "x axis interval": Array [
-    "Jan 1",
-    "Jan 3",
-    "Jan 5",
+  "chartData": Array [
+    Object {
+      "tooltip": "5 t('curiosity-graph.tooltipLabel') Dec 31",
+      "x": 0,
+      "xAxisLabel": "Dec 31",
+      "y": 5,
+    },
+    Object {
+      "tooltip": "7 t('curiosity-graph.tooltipLabel') Jan 1 2019
+ +2 t('curiosity-graph.tooltipPreviousLabelDaily')",
+      "x": 1,
+      "xAxisLabel": "Jan 1 2019",
+      "y": 7,
+    },
   ],
+  "chartXAxisLabelIncrement": 5,
 }
 `;
 
 exports[`GraphHelpers should have specific functions: graphHelpers 1`] = `
 Object {
-  "chartDateFormat": "MMM D",
-  "convertGraphUsageData": [Function],
-  "getChartDomain": [Function],
-  "getTickValues": [Function],
-  "zeroedUsageData": [Function],
+  "convertChartData": [Function],
+  "fillFormatChartData": [Function],
+  "getChartXAxisLabelIncrement": [Function],
+  "getGranularityDateType": [Function],
+  "getGraphLabels": [Function],
+  "getLabel": [Function],
+}
+`;
+
+exports[`GraphHelpers should return a date type based on granularity: granularity date type 1`] = `
+Object {
+  "daily": "days",
+  "monthly": "months",
+  "quarterly": "months",
+  "weekly": "weeks",
+}
+`;
+
+exports[`GraphHelpers should return a label based on granularity: granularity based label 1`] = `
+Object {
+  "daily": Object {
+    "label": "t('curiosity-graph.tooltipLabel')",
+    "previousLabel": "t('curiosity-graph.tooltipPreviousLabelDaily')",
+  },
+  "monthly": Object {
+    "label": "t('curiosity-graph.tooltipLabel')",
+    "previousLabel": "t('curiosity-graph.tooltipPreviousLabelMonthly')",
+  },
+  "quarterly": Object {
+    "label": "t('curiosity-graph.tooltipLabel')",
+    "previousLabel": "t('curiosity-graph.tooltipPreviousLabelQuarterly')",
+  },
+  "weekly": Object {
+    "label": "t('curiosity-graph.tooltipLabel')",
+    "previousLabel": "t('curiosity-graph.tooltipPreviousLabelWeekly')",
+  },
+}
+`;
+
+exports[`GraphHelpers should return a x axis tick increment based on granularity: x axis tick increment 1`] = `
+Object {
+  "daily": 5,
+  "monthly": 1,
```

Comment:
this is really nice! Custom increments by granularity...

These settings seem to be OK.. I would possibly consider increasing monthly to 2 though for better mobile spacing... it's legible though.
![Screen Shot 2019-08-22 at 10 59 54 AM](https://user-images.githubusercontent.com/4237045/63525725-2682f580-c4cc-11e9-921d-f25fafe0f360.png)

Reply:
Debating on whether to iterate on this facet. Part of the graph helpers solution is temporary, would have to start tracking tick increment against label/year display to handle this request

Reply:
- pushed an update so the tick increment affects the displayed year
- pushed an update so months are now every 2

![Screen Shot 2019-08-22 at 1 14 55 PM](https://user-images.githubusercontent.com/3761375/63535184-dd886c80-c4de-11e9-9aed-7c4214a1e5af.png)

Reply:
Lookin good!
I think the colors/styling might get tweaked but we can merge that bit in later.

Reply:
awesome!